tutorial/opt/aero/plot_optHist.py

```diff
@@ -3,49 +3,92 @@
 import matplotlib.pyplot as plt
 from pyoptsparse import History
 
-plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
-plt.rcParams["font.family"] = "serif"
-plt.rcParams["font.size"] = 20
-plt.rcParams["xtick.labelsize"] = 16
-plt.rcParams["ytick.labelsize"] = 16
-plt.rcParams["lines.linewidth"] = 3.0
 
-parser = argparse.ArgumentParser()
-parser.add_argument("--histFile", type=str, default="opt.hst")
-parser.add_argument("--outputDir", type=str, default="./")
-args = parser.parse_args()
+def main():
 
-optHist = History(args.histFile)
-histValues = optHist.getValues()
+    plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
+    plt.rcParams["font.family"] = "serif"
+    plt.rcParams["font.size"] = 20
+    plt.rcParams["xtick.labelsize"] = 16
+    plt.rcParams["ytick.labelsize"] = 16
+    plt.rcParams["lines.linewidth"] = 3.0
 
-fig, axes = plt.subplots(nrows=5, sharex=True, constrained_layout=True, figsize=(14, 10))
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--histFile", type=str, default="opt.hst")
+    parser.add_argument("--opt", type=str, default="SLSQP", choices=["IPOPT", "SLSQP", "SNOPT"])
+    parser.add_argument("--outputDir", type=str, default="./")
+    args = parser.parse_args()
 
-axes[0].plot("nMajor", "optimality", data=histValues, label="Optimality")
-axes[0].plot("nMajor", "feasibility", data=histValues, label="Feasibility")
-axes[0].set_yscale("log")
-axes[0].axhline(1e-4, linestyle="--", color="gray")
-axes[0].annotate("Convergence Criteria", xy=(3, 9e-5), ha="left", va="top", fontsize=24, color="gray")
-axes[0].legend(fontsize=20, labelcolor="linecolor", loc="upper right", frameon=False)
-axes[0].autoscale(enable=True, tight=True)
+    try:
+        if not os.path.isfile(args.histFile):
+            raise FileNotFoundError(f"History file '{args.histFile}' not found.")
 
-axes[1].plot("nMajor", "obj", data=histValues, label="Objective")
-axes[1].set_ylabel("Objective \n($C_D$)", rotation="horizontal", ha="right", fontsize=24)
+        optHist = History(args.histFile)
+        histValues = optHist.getValues()
 
-axes[2].plot("nMajor", "alpha_fc", data=histValues, label="Alpha")
-axes[2].set_ylabel(r"$\alpha [^\circ]$", rotation="horizontal", ha="right", fontsize=24)
+        if args.opt == "SNOPT":
```

Comment:
It seems most of the plotting is the same between the two blocks, only difference is SNOPT has feasibility/optimality in `axes[0]` and there are 5 subplots. You can probably consolidate this so there is no code duplication. E.g.

```
if SNOPT:
    # make 5 subplots
    # plot optimality/feasibility on axes[00
else:
    # make 4 subplots
# plot the rest on axes[-4:]
```

Reply:
This is also a good idea

Reply:
I have removed code duplication by:
    if SNOPT:
        # make 5 subplots
        # plot optimality/feasibility on axes[4]
    else:
        # make 4 subplots
    # plot the rest on axes[-4:]